### PR TITLE
Implement mload with alignment correction built-in.

### PR DESCRIPTION
**Pull request by chriseth**
Depends on #720

### CODE REVIEW

**Comment by chriseth** [comment]
This moves the keccak loop start from 22100 to 18700.
Also performance goes from 4.4 rows/sec to 5k rows/sec (because it does not need the binary machine any more for most memory accesses).
What is a bit strange is that (after optimization) the witness col count goes from 271 up to 272, the fixed from 212 to 216.

**Comment by leonardoalt** [comment]
needs a rebase maybe?

File: riscv/src/compiler.rs [hunk 1]
```diff
@@ -1274,8 +1274,7 @@ fn process_instruction(instr: &str, args: &[Argument], coprocessors: &CoProcesso
             vec![
                 format!("tmp1 <== wrap({rd} + {off});"),
                 "addr <== and(tmp1, 0xfffffffc);".to_string(),
```
Comment (leonardoalt) [inline riscv/src/compiler.rs]: leftover `addr`?

File: riscv/src/compiler.rs [hunk 2]
```diff
@@ -1257,8 +1258,7 @@ fn process_instruction(instr: &str, args: &[Argument], coprocessors: &CoProcesso
...
```
Comment (leonardoalt) [inline riscv/src/compiler.rs]: leftover addr

**Review by leonardoalt — COMMENTED** [review]
maybe remove the `addr` register?

**Comment by leonardoalt** [comment]
Ah you did remove it, I was reviewing commit by commit because of the merge commits

File: riscv/src/compiler.rs [hunk 3]
```diff
@@ -521,6 +519,8 @@ fn preamble(coprocessors: &CoProcessors) -> String {
         // { W, X, Z} in { shr.value, shr.amount, shr.amount}
     }
 
+    /// Stores Z at addres Y % 2**32. Y can be between 0 and 2**33.
```
Comment (leonardoalt) [inline riscv/src/compiler.rs:522]: s/addres/address

**Comment by leonardoalt** [comment]
Looks good! Just a typo left. It is weird though that there are more columns now

**Review by leonardoalt — APPROVED** [review]

**Pull request merged**